### PR TITLE
CEM-773 SegmentedButton

### PR DESCRIPTION
**Pull request by Ampatte2**
A segmented button that takes an array of segment types and creates a row of buttons.

CEM-773-SegmentedButton

CEM-773-SegmentedButton - More
CEM-773-SegmentedButton - Less

### CODE REVIEW

**Review by ralph-dev — APPROVED** [review]
LGTM, a bit weird you override the onClick but must be for good reason?

File: src/Inputs/SegmentedButton.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    onClick,
+    ...props
+}): React.ReactElement => {
+
+    const renderSegments = useCallback(() => {
+        return segments.map((segment, index)=> {
+            return (
+                <Segment 
+                    active={segment.active}
+                    onClick={(event)=> onClick(event, index)}
```
Comment (ralph-dev) [inline src/Inputs/SegmentedButton.tsx:39]:
```suggestion
                    onClick={(event) => onClick(event, index)}
```

File: src/Inputs/SegmentedButton.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        height: ${height};
+    `}
+`
+const Segment = styled.div<ISegmentProps>`
+    cursor: pointer;
+    ${transition(['background-color', 'opacity'])}
+    ${flex('center')}
+    ${clickable('#ffffff', 0.05)};
+    ${({theme, active})=> {
+        const style = active ? `background-color: ${theme.colors.primary}; color: ${theme.colors.background};` : ` background-color: ${theme.colors.background}; color: ${theme.colors.text};`
```
Comment (ralph-dev) [inline src/Inputs/SegmentedButton.tsx:94]: what is this lmfao

Comment (ralph-dev) [reply src/Inputs/SegmentedButton.tsx:94]: Have you tried 
CSS Variants :)

**Comment by Ampatte2** [comment]
Wanted to expose the index of the array, and in the past have learned that having access to the event information is always a plus.

File: src/Inputs/SegmentedButton.tsx (hunk 2 above)
Comment (ralph-dev) [reply src/Inputs/SegmentedButton.tsx:94]: image

Comment (Ampatte2) [reply src/Inputs/SegmentedButton.tsx:94]: I thought was originally changing clickable, and have had troubles with interpolating the mixins in the past. I will revert to just the ternary

**Pull request merged**